test/unit/rules/require-valid-form-groups-test.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+
+const ERROR_MESSAGE = require('../../../lib/rules/require-valid-form-groups').ERROR_MESSAGE;
+
+generateRuleTests({
+  name: 'require-valid-form-groups',
+
+  config: true,
+
+  good: [
+    `<fieldset>
+  <legend>Preferred Mascot Version</legend>
+      <div>
+            <label for="radio-001">Chicago Zoey</label>
+                  <input id="radio-001" type="radio" name="prefMascot-Zoey" value="chicago zoey" />
+                    </div>
+                      <div>
+                            <label for="radio-002">Office Hours Tomster</label>
+                                  <input id="radio-002" type="radio" name="prefMascot-OfficeHoursTomster" value="office hours tomster" />
+                                    </div>
+                                      <div>
+          <label for="radio-003">A11y Zoey</label>
+          <input id="radio-003" type="radio" name="prefMascot-Zoey" value="a11y zoey" />
+      </div>
+    </fieldset>`,
+    `<div role="group" aria-labelledby="preferred-mascot-heading">
+  <div id="preferred-mascot-heading">Preferred Mascot Version</div>
+      <label for="radio-001">Chicago Zoey</label>
+        <input id="radio-001" type="radio" name="prefMascot-Zoey" value="chicago zoey" />
+          <label for="radio-002">Office Hours Tomster</label>
+            <input id="radio-002" type="radio" name="prefMascot-OfficeHoursTomster" value="office hours tomster" />
+              <label for="radio-003">A11y Zoey</label>
+      <input id="radio-003" type="radio" name="prefMascot-Zoey" value="a11y zoey" />
+  </div>`,
+  ],
+
+  bad: [
```

Comment:
These bad tests may need a little more work- we want to catch that there are multiple elements that should be grouped but aren't.